### PR TITLE
Psalaberria002 hyperunique

### PR DESCRIPTION
**Pull request by camsaul**
PR #5227 with merge conflicts fixed

### CODE REVIEW

File: src/metabase/driver/druid/query_processor.clj [hunk 1]
```diff
@@ -217,6 +218,16 @@
 
 (defn- ag:filtered  [filtr aggregator] {:type :filtered, :filter filtr, :aggregator aggregator})
 
+(defn- ag:distinct [field output-name]
+  (if (= (:base-type field) :type/DruidHyperUnique)
```
Comment (senior) [inline src/metabase/driver/druid/query_processor.clj]: Doesn't matter right now in the codebase, but it seems like this should be `isa?` for the future.

File: src/metabase/driver/druid/query_processor.clj [hunk 2]
```diff
@@ -217,6 +218,16 @@
...
+    {:type      :hyperUnique
+     :name      output-name
+     :fieldName (->rvalue field)}
+    {:type       :cardinality
+     :name       output-name
+     :fieldNames [(->rvalue field)]}))
+
```
Comment (senior) [inline src/metabase/driver/druid/query_processor.clj]: Extra line

File: src/metabase/driver/druid.clj [hunk 3]
```diff
@@ -70,17 +70,31 @@
...
+                     "dataSource"    datasource
+                     "intervals"     ["1999-01-01/2114-01-01"]
+                     "analysisTypes" []
+                     "merge"         true}))
+
+(defn- druid-type->base-type [field-type]
+  (case field-type
+    "STRING"      :type/Text
+    "FLOAT"       :type/Float
+    "LONG"        :type/Float
```
Comment (senior) [inline src/metabase/driver/druid.clj]: Shouldn't this be `:type/Long`?

File: src/metabase/driver/druid.clj [hunk 4]
```diff
@@ -70,17 +70,31 @@
...
+    :type/Float))
+
+(defn- describe-table-field [field-name {field-type :type, :as info}]
   ;; all dimensions are Strings, and all metrics as JS Numbers, I think (?)
   ;; string-encoded booleans + dates are treated as strings (!)
-  (assoc (case druid-field-type
-           :metric    {:database-type "metric",    :base-type :type/Float}
-           :dimension {:database-type "dimension", :base-type :type/Text})
-    :name field-name))
+  {:name          (name field-name)
```
Comment (senior) [inline src/metabase/driver/druid.clj]: Nit picky, but seems like `name` here is not needed.

**Review by senior — COMMENTED** [review]
Overall 👍. Just want to double check the "LONG" -> `:type/Float` change

File: src/metabase/driver/druid.clj (hunk 3 above)
Comment (camsaul) [reply src/metabase/driver/druid.clj]: That's a good question. @psalaberria002 ?

**Comment by camsaul** [comment]
I pinged @psalaberria002 (author of the original PR) about the `:type/Long` thing but I'll clean up the other stuff

**Review by senior — APPROVED** [review]

File: src/metabase/driver/druid/query_processor.clj (hunk 2 above)
Comment (camsaul) [reply src/metabase/driver/druid/query_processor.clj]: fixed

File: src/metabase/driver/druid.clj (hunk 4 above)
Comment (camsaul) [reply src/metabase/driver/druid.clj]: turns out it is needed :)

File: src/metabase/driver/druid.clj (hunk 3 above)
Comment (psalaberria002) [reply src/metabase/driver/druid.clj]: Hmm. Long time since I checked. I am not sure why I did that.
I guess ir should be `:type/Long`.

Comment (camsaul) [reply src/metabase/driver/druid.clj]: `:type/Integer` probably makes the most sense. We don't currently differentiate between 64-bit and 32-bit integers for query-writing purposes which is why there's no separate `:type/Long` yet. `:type/BigInteger` however we do use in places where things specifically map to a java `BigInteger` type. I'll change this

**Comment by psalaberria002** [comment]
Nice! Thank you.

**Pull request merged**